src/ast/annot.ml

```diff
@@ -14,6 +16,24 @@ and item =
 
 module NameMap = Map.Make (String)
 
+(* TODO: find a better way to format annotations, possibly by
+      - recording extra format information when parsing
+      - defining rules specific to each sort of annotations *)
+
+let rec pp_annot fmt annot =
+  pp fmt "(annot %a@\n  @[<b 2>%a@]@\n)" pp_string annot.annotid
+    (pp_list ~pp_sep:pp_space pp_item)
+    annot.items
+
+and pp_item fmt = function
+  | Atom atom -> pp_string fmt atom
+  | String str -> pp_string fmt str
+  | Id id -> Types.pp_id fmt id
+  | Int i -> pp_string fmt i
+  | Float f -> pp_string fmt f
+  | Parens items -> pp_list ~pp_sep:pp_space pp_item fmt items
+  | Annot annot -> pp_annot fmt annot
```

Comment:
You'll have to replace `pp_string` by `Fmt.string`, `pp_list` by `list` and `~pp_sep` by `~sep`.